### PR TITLE
Reference `clippy_utils` docs on nightly-rustc and some other documentation updates

### PR DESCRIPTION
**Pull request by xFrednet**
The `clippy_utils` crate is now part of the nightly-rustc documentation. See [**very beautiful documentation**](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/). This PR references them in our documentation and updates some other documentation. 

changelog: none

### CODE REVIEW

File: doc/common_tools_writing_lints.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ You may need following tooltips to catch up with common operations.
 
 - [Common tools for writing lints](#common-tools-for-writing-lints)
   - [Retrieving the type of an expression](#retrieving-the-type-of-an-expression)
-  - [Checking if an expression is calling a specific method](#checking-if-an-expr-is-calling-a-specific-method)
+  - [Checking if an expr is calling a specific method](#checking-if-an-expr-is-calling-a-specific-method)
+  - [Checking for a specific type](#checking-for-a-specific-type)
   - [Checking if a type implements a specific trait](#checking-if-a-type-implements-a-specific-trait)
   - [Checking if a type defines a specific method](#checking-if-a-type-defines-a-specific-method)
   - [Dealing with macros](#dealing-with-macros)
```
Comment (xFrednet) [inline doc/common_tools_writing_lints.md:11]: The table of content was automatically updated by one of my IDE add-ons (IDK which one ^^). This PR also makes the previous `#` headlines to `##` this makes them a bit smaller and keeps the table of content consistent. :upside_down_face:

Comment (xFrednet) [inline doc/common_tools_writing_lints.md:11]: The table of content was automatically updated by one of my IDE add-ons (IDK which one ^^). This PR also makes the previous `#` headlines to `##` headlines. This makes them a bit smaller and keeps the table of content consistent. :upside_down_face:

File: doc/common_tools_writing_lints.md [hunk 2]
```diff
@@ -74,7 +77,44 @@ impl LateLintPass<'_> for MyStructLint {
...
+        if is_type_diagnostic_item(cx, ty, sym::Option) {
+            // The type is an `Option`
+        }
+
+        // 2. Using lang items
+        if is_type_lang_item(cx, ty, LangItem::RangeFull) {
+            // The type is a full range like `.drain(..)`
+        }
+
+        // 3. Using the type path
```
Comment (flip1995) [inline doc/common_tools_writing_lints.md:107]: I would add a note here, that we're trying to avoid doing this as much as possible.

**Review by flip1995 — APPROVED** [review]
Just a small note. r=me with this change.

**Comment by xFrednet** [comment]
@bors r=flip1995

:tada:

File: doc/common_tools_writing_lints.md (hunk 2 above)
Comment (xFrednet) [reply doc/common_tools_writing_lints.md:107]: I also added this to the *Checking if a type implements a specific trait* documentation

**Pull request merged**